### PR TITLE
[export] Add setattr for ep.example_inputs

### PR DESCRIPTION
**Pull request by angelayi**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #140991
* __->__ #140990

Differential Revision: [D66136725](https://our.internmc.facebook.com/intern/diff/D66136725)

### CODE REVIEW

**Comment by angelayi** [comment]
@angelayi has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D66136725).

File: test/export/test_export.py [hunk 1]
```diff
@@ -3537,6 +3537,48 @@ def forward(self, arg1, arg2, a=1, b=2):
...
+        ep.example_inputs = (
+            (torch.ones(3),),
+            {"x": torch.ones(3), "z": torch.ones(3), "y": torch.ones(3)},
+        )
+        ep.module()(*ep.example_inputs[0], **ep.example_inputs[1])
+
+        with self.assertRaisesRegex(ValueError, "Example inputs should be a tuple"):
+            ep.example_inputs = (torch.ones(3),)
+
+        with self.assertRaisesRegex(ValueError, "Ran into a kwarg keyword mismatch"):
```
Comment (yushangdi) [inline test/export/test_export.py:3573]: does it work when the keyword has default values? e.g. `def forward(self, a, *, x, y, z=torch.ones(2))`. Now am I still required to provide a value for `z`?

Comment (angelayi) [reply test/export/test_export.py:3573]: Lol just found out that dynamo does not support default tensor kwargs:
> torch._dynamo.exc.Unsupported: Tracing through optional input is not supported yet
But for non-strict, we will not treat the kwarg with a default value as an input, so you *cant* provide a value for z.

Comment (yushangdi) [reply test/export/test_export.py:3573]: > Lol just found out that dynamo does not support default tensor kwargs:
> 
> > torch._dynamo.exc.Unsupported: Tracing through optional input is not supported yet
> > But for non-strict, we will not treat the kwarg with a default value as an input, so you _cant_ provide a value for z.

lol I see.

**Review by yushangdi — APPROVED** [review]

**Review by ydwu4 — APPROVED** [review]

**Comment by angelayi** [comment]
@angelayi has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D66136725).

**Pull request closed without merging**